roles/foreman_proxy/tasks/main.yml

```diff
@@ -46,6 +51,79 @@
     content: "{{ foreman_proxy_client_ca }}"
     dest: /etc/foreman-proxy/foreman_ssl_ca.pem
 
+- name: 'Create identity directory'
+  file:
+    path: "{{ foreman_proxy_remote_execution_ssh_dir }}"
+    state: directory
+    mode: '0700'
+    owner: foreman-proxy
+    group: foreman-proxy
+
+- name: 'Symlink proxy home .ssh to identity directory'
+  file:
+    dest: "/usr/share/foreman-proxy/.ssh"
+    src: "{{ foreman_proxy_remote_execution_ssh_dir }}"
+    owner: foreman-proxy
+    group: foreman-proxy
+    state: link
+
+- name: Generate /etc/ssh/ RSA host key
+  command: 'ssh-keygen -q -t rsa -b 4096 -f {{ foreman_proxy_remote_execution_ssh_dir }}/{{ foreman_proxy_remote_execution_ssh_keypair_name }} -C "Foreman Remote execuction key" -N ""'
```

Comment:
Could switch to Ansible module -- https://docs.ansible.com/ansible/devel/modules/openssh_keypair_module.html

Reply:
Would also let you drop the next two file tasks

Reply:
Welp, I see further down below you already had that thought.

Reply:
Maybe move it before these command so thats more obvious reading top-bottom